src/main/java/gregtech/common/items/behaviors/TerminalBehaviour.java

```diff
@@ -59,7 +59,7 @@ public ActionResult<ItemStack> onItemRightClick(World world, EntityPlayer player
     @Override
     public void onUpdate(ItemStack itemStack, Entity entity) {
         NBTTagCompound tabletNBT = itemStack.getOrCreateSubCompound("terminal");
-        if (tabletNBT.hasKey("_ar")) {
+        if (tabletNBT.hasKey("_ar") && entity.ticksExisted % 20 == 0) {
```

Comment:
This implementation is correct, but if you write that `if` check other way around it will be less expensive, as searching for key in hash map is quite costly operation.
```suggestion
        if (entity.ticksExisted % 20 == 0 && tabletNBT.hasKey("_ar")) {
```